setup.py

```diff
@@ -47,6 +47,7 @@ def get_identifier(json):
     name="jupyter-repo2docker",
     version=versioneer.get_version(),
     install_requires=[
+        "boto3",
```

Comment:
Should boto3 be a hard requirement? I would expect close to zero local users would use this feature.

Reply:
I don't think it should be. Wondering if there is a smaller/more lightweight library that implements the S3 part as that is all we need.

Reply:
I'll make it optional. I came across https://github.com/smore-inc/tinys3 but doesn't look active.

It's possible to do this using tornado/requests, the main difficulty is calculating the S3v4 signature: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

The dependencies of botocore don't look too bad:
https://github.com/boto/botocore/blob/717eb38a99d0f3eed9648abd62184fbc6a7947ef/setup.py#L26-L36
```
jmespath>=0.7.1,<1.0.0
python-dateutil>=2.1,<3.0.0
urllib3>=1.20,<1.26
```
It's possible to make the S3 request using botocore. For boto3 which adds a higher level wrapper add
https://github.com/boto/boto3/blob/73865126cad3938ca80a2f567a1c79cb248169a7/setup.py#L16-L20
```
s3transfer>=0.3.0,<0.4.0
```

The other library I'm aware of is Minio's python client, though I haven't used it:
https://github.com/minio/minio-py/blob/b67bba368dfc244403b73538f6a2eeafe2016155/setup.py#L40-L46
```
urllib3
pytz
certifi
python-dateutil
configparser
```